### PR TITLE
Exit when ID overflow occurs in a fuzzing build

### PR DESCRIPTION
**Pull request by afd**
Currently if an ID overflow occurs, spirv-opt (and other users of
IRContext) emits a warning and starts returning 0 when fresh ids are
requested. This tends to lead to crashes - such as null pointer
exceptions. When these arise during fuzzing they lead to auto-reported
bugs.

This change uses an ifdef guard to instead gracefully exit as soon as an
ID overflow occurs when the build is a fuzzing build.

Related issue: #4539.

### CODE REVIEW

File: source/opt/ir_context.h [hunk 1]
```diff
@@ -518,6 +518,14 @@ class IRContext {
         std::string message = "ID overflow. Try running compact-ids.";
         consumer()(SPV_MSG_ERROR, "", {0, 0, 0}, message.c_str());
       }
+#ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
+      // If TakeNextId returns 0, it is very likely that execution will
+      // subsequently fail. Such failures are false alarms from a fuzzing point
+      // of view: they are due to the fact that too many ids were used, rather
+      // than being due to an actual bug. Thus, during a fuzzing build, it is
+      // preferable to bail out when ID overflow occurs.
+      exit(0);
```
Comment (s-perron) [inline source/opt/ir_context.h:527]: Did we confirm that we cannot exit with an error code of some type?

**Review by s-perron — APPROVED** [review]
LGTM.  However, I would prefer that we exit with a value other than 0 if we can.

File: source/opt/ir_context.h (hunk 1 above)
Comment (afd) [reply source/opt/ir_context.h:527]: I'll double check that.

Comment (afd) [reply source/opt/ir_context.h:531]: I got confirmation that a non-zero exit code will indeed be regarded as a crash, so we need to stick with exit(0) here. I have elaborated on this in the comment.

Comment (s-perron) [reply source/opt/ir_context.h:531]: Okay.

**Review by s-perron — APPROVED** [review]

**Pull request merged**